core-web/libs/sdk/client/src/lib/client/page/page-api.ts

```diff
@@ -0,0 +1,441 @@
+     *         BlogCollection(limit: 3) {
+     *           title
+     *           urlTitle
+     *           publishDate
+     *         }
+     *       }
+     *     `
+     *   },
+     *   nav: {
+     *     mainNav: `
+     *       query MainNav {
+     *         Nav(identifier: "main-nav") {
+     *           title
+     *           items {
+     *             label
+     *             url
+     *           }
+     *         }
+     *       }
+     *     `
+     *   }
+     * });
+     * ```
+     */
+    get(url: string, options?: PageRequestParams): Promise<DotCMSPageAsset>;
+    get(url: string, options?: GraphQLPageOptions): Promise<DotCMSGraphQLPageResponse>;
+    get(
+        url: string,
+        options?: PageRequestParams | GraphQLPageOptions
+    ): Promise<DotCMSPageAsset | DotCMSGraphQLPageResponse> {
+        if (!options) {
+            return this.#getPageFromAPI(url);
+        }
+
+        if (this.#isGraphQLRequest(options)) {
+            return this.#getPageFromGraphQL(url, options);
+        }
+
+        if (this.#isPageRequestParams(options)) {
+            return this.#getPageFromAPI(url, options);
+        }
+
+        console.warn('Invalid options provided to get method:', options);
+
+        return this.#getPageFromAPI(url);
+    }
+
+    /**
+     * Determines if the provided options object is for a GraphQL request.
+     *
+     * @param {PageRequestParams | GraphQLPageOptions} options - The options object to check
+     * @returns {boolean} True if the options are for a GraphQL request, false otherwise
+     * @internal
+     */
+    #isGraphQLRequest(
+        options: PageRequestParams | GraphQLPageOptions
+    ): options is GraphQLPageOptions {
+        const isPageRequestParams = this.#isPageRequestParams(options);
+
+        if (isPageRequestParams) {
+            return false;
+        }
+
+        return !!options?.['query'];
+    }
+
+    /**
+     * Type guard that checks if the provided options object matches the PageRequestParams interface.
+     *
+     * @param {unknown} options - The options object to check
+     * @returns {boolean} True if the options match PageRequestParams, false otherwise
+     * @internal
+     *
+     * @example
+     * ```typescript
+     * const options = {
+     *   siteId: '123',
+     *   mode: 'PREVIEW_MODE',
+     *   languageId: 1
+     * };
+     *
+     * if (this.#isPageRequestParams(options)) {
+     *   // options is typed as PageRequestParams
+     * }
+     * ```
+     */
+    #isPageRequestParams(options: unknown): options is PageRequestParams {
+        if (typeof options !== 'object' || options === null) {
+            return false;
+        }
+
+        const validKeys = [
+            'siteId',
+            'mode',
+            'languageId',
+            'personaId',
+            'fireRules',
+            'depth',
+            'publishDate'
+        ];
+
+        return Object.keys(options).every((key) => validKeys.includes(key));
```

Comment:
This will fail if I wrongly send more `queryParams` so it probably will fail inside UVE, because we send experiments and date params

Reply:
A `some` instead of `every` should do the trick

Reply:
By checking with `some` method, we should be safe